### PR TITLE
feat: namananand/ins 3273 automation of connector and operator docs

### PR DESCRIPTION
**Pull request by iamnamananand996**
Because

- automation of connector and operator docs

This commit

- automation of connector and operator docs

### CODE REVIEW

File: update_connector_operetor.sh
Comment (jvallesm) [inline update_connector_operetor.sh:1]: typo: `update_connector_operator.sh`

File: update_connector_operetor.sh [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    local lang=$2
+
+    # Find all config directories for the given component type
+    local component_dirs=$(find component/pkg/${component_type} -type d -name "config")
+
+    for component_dir in $component_dirs; do
+        local component_name=$(basename $(dirname $(dirname ${component_dir})))
+
+        # Generate README and copy files for English
+        echo compogen readme ${component_dir} component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx --${component_type}
```
Comment (jvallesm) [inline update_connector_operetor.sh:31]: It'd be better if we didn't rely on the `v0` bit, otherwise this script will break when we introduce a new major version.

File: update_connector_operetor.sh [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        compogen readme ${component_dir} component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx --${component_type}
+        echo cp component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx docs/vdp/${component_type}s/${component_name}.$lang.mdx
+        cp component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx docs/vdp/${component_type}s/${component_name}.$lang.mdx
```
Comment (jvallesm) [inline update_connector_operetor.sh:34]: ❓ Why copying instead of setting the output of the command to the final directory?

File: update_connector_operetor.sh [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    # Find all config directories for the given component type
+    local component_dirs=$(find component/pkg/${component_type} -type d -name "config")
+
+    for component_dir in $component_dirs; do
+        local component_name=$(basename $(dirname $(dirname ${component_dir})))
+
+        # Generate README and copy files for English
+        echo compogen readme ${component_dir} component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx --${component_type}
+        compogen readme ${component_dir} component/pkg/${component_type}/${component_name}/v0/${component_name}.$lang.mdx --${component_type}
```
Comment (jvallesm) [inline update_connector_operetor.sh:32]: Are we generating the same file with 2 different names (`en` and `zh_CN`)?  Why not copying the file directly? Or why having different languages only nominally? Don't we have a way to fallback to the default language if a document doesn't have a translated version?

File: update_connector_operetor.sh [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+process_component_type "operator" "zh_CN"
+
+# <----- CONNECTOR ------>
+
+# compogen readme component/pkg/connector/airbyte/v0/config/ component/pkg/connector/airbyte/v0/airbyte.en.mdx --connector
+# cp component/pkg/operator/airbyte/v0/airbyte.en.mdx docs/vdp/operators/airbyte.en.mdx
+# compogen readme component/pkg/connector/airbyte/v0/config/ component/pkg/connector/airbyte/v0/airbyte.zh_CN.mdx --connector
+# cp component/pkg/operator/airbyte/v0/airbyte.zh_CN.mdx docs/vdp/operators/airbyte.zh_CN.mdx
+
+compogen readme component/pkg/connector/archetypeai/v0/config/ component/pkg/connector/archetypeai/v0/archetype-ai.en.mdx --connector
```
Comment (jvallesm) [inline update_connector_operetor.sh:49]: The whole point about adding [build tags](https://github.com/instill-ai/component/blob/main/pkg/operator/base64/v0/main.go#L1C1-L1C63) is that you can spare all the generation and do `go generate -run compogen ./...` (`-run` limits generation to `compogen` tags only).

(Re-)Generating all the docs in a single command and copying them to the target directory will produce a much simpler script:

```sh
go generate -run compogen ./components/...
for doc in $(find component -name README.mdx); do
  component_type=$(echo $doc | cut -d'/' -f 3); component_name=$(echo $doc | cut -d'/' -f 4)
  target_en=docs/vdp/${component_type}s/$component_name.en.mdx
  target_cn=docs/vdp/${component_type}s/$component_name.zh_CN.mdx
  cp $doc $target_en
  cp $doc $target_cn
done;
```

Instead of using `component_type`, if we want to keep the current directory separation in the destination, you should write a function that maps `component_name` to the target directory (operators, data connectors, etc.).

File: update_connector_operetor.sh (hunk 3 above)
Comment (iamnamananand996) [reply update_connector_operetor.sh:32]: Hi @jvallesm, in documentation we support 2 language for now, English and Chinese, that's why we have 2 names.

> Don't we have a way to fallback to the default language if a document doesn't have a translated version?

good suggestion, correctly we don't have this, will add this kind of redirection.

File: update_connector_operator.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            echo "app-connectors"
+            ;;
+        base64|end|image|json|start|text)
+            echo "operators"
+            ;;
+    esac
+}
+
+# Generate components
+go generate -run compogen ./component/...
```
Comment (jvallesm) [inline update_connector_operator.sh:31]: With https://github.com/instill-ai/component/pull/80 this (and lines 3-4) will be `make build-doc && make gen-doc`

Comment (jvallesm) [reply update_connector_operator.sh:31]: Again, we should aim to remove generation from this script and delegate on a GH action on the component repo that runs the generation command when the definition files change.

Updating the generated files here as a first step is valid ✅ But then let's have a ticket for the automation of the README generation.

File: update_connector_operator.sh
Comment (jvallesm) [inline update_connector_operator.sh:1]: ❓ Is this script meant to be run manually? If so, it's probably worth a line in the repo's README or CONTRIBUTING docs.

Comment (iamnamananand996) [reply update_connector_operator.sh:1]: Updated

File: .github/DOCS-CONTRIBUTING.md [hunk 6]
```diff
@@ -67,6 +67,13 @@ There have three content folders and their corresponding page, assets folder.
 
 To be documented...
 
+
+### Auto genrate connector and operator docs
```
Comment (jvallesm) [inline .github/DOCS-CONTRIBUTING.md:71]:
```suggestion
### Auto generate connector and operator docs
```

Comment (jvallesm) [reply .github/DOCS-CONTRIBUTING.md:71]: Once this is fixed the PR looks good 👍

**Review by jvallesm — APPROVED** [review]

File: docs/vdp/operators/end.en.mdx
Comment (jvallesm) [inline docs/vdp/operators/end.en.mdx:1]: Why did you add the start and end documents as generated by `compogen`? The former version is clearly better:

- start and end shouldn't be documented as operators but components
- the new version has TODO sections
- the new section only list the task the component can perform, which is useless as documentation.

Comment (iamnamananand996) [reply docs/vdp/operators/end.en.mdx:1]: Hi @jvallesm , it have just excuted the `update_connector_operator.sh` script for final merge, it get automatically generated, updated back to old one

**Review by jvallesm — APPROVED** [review]
